Add explanation about aggregate identifiers and uniqueness (#3324)
AxonFramework/AxonFramework: docs/old-reference-guide/modules/axon-framework-commands/pages/modeling/aggregate.adoc

Aggregates are commonly used to model the core business entities in a system based on link:https://www.axoniq.io/concepts/domain-driven-design[Domain-Driven Design] and link:https://www.axoniq.io/concepts/cqrs-and-event-sourcing[CQRS].
This chapter will cover the basics on how to implement one in Axon Framework.

== Aggregate identifiers

Each aggregate has an identifier that uniquely identifies it within the context.
Events published from this aggregate will contain this identifier,
so that Axon Framework can later load the aggregate from its event stream.

Commands that target aggregates contain a reference to this identifier.
Axon Framework will load the events for the aggregate with this identifier,

> replay them on an empty instance, and invoke the command.

Reviewer: Perhaps add one sentence to conclude that this behavior of stream retrieval, empty instance creations, and replaying of the events, is event sourcing in action?
```suggestion
replay them on an empty instance, and invoke the command. In doing so, this style supports the event sourcing of aggregates.
```

Author: Agreed, but I modified it a little bit :)

[TIP]
====

> The aggregate identifier has to be unique in your event store.
> Events are only loaded based on the identifier, not
> the type of the aggregate.
> This means that if you have two aggregate types with the same identifier, Axon will
> load the events for both aggregates when handling a command.
> Mixing these events might lead to unexpected behavior, so design
> your aggregate identifiers with uniqueness in mind.

Reviewer: ```suggestion
The aggregate identifier has to be globally unique in your event store.
This means that events are loaded based only on the identifier, and nothing else.
Henceforth, the aggregate's type does not influence uniqueness at all.
This means that if you have two aggregate types with the same identifier, Axon will
load the events for both aggregates when handling a command.
Mixing these events might lead to unexpected behavior, so we strongly recommend to design
your aggregate identifiers with uniqueness in mind.
```

====

Note that commands that construct a new aggregate (via a constructor) do not need an identifier.
However, your `@AggregateIdentifier` field needs to have a value after the first command,

> or Axon Framework will not know which aggregate identifier to publish the events to.

Reviewer: ```suggestion
or Axon Framework will not know which aggregate identifier to publish the events for.
```
